Card style updated, created a single shared component [Fixes #2251]

Anuragtech02 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
- [x] Updated card style on https://ethereum.org/en/contributing/translation-program/ to match the one on the Languages page: https://ethereum.org/en/languages/
- [x]  Created a single shared component in SharedStyledComponents.js
<!--- Describe your changes in detail -->

Created a single shared component CardItem in SharedStyledComponents and replaced the cards on `/languages` and `/about` ( #2076 & #2195 ) with the new shared component. Doing this also unified card hover transition.

## Related Issue #2251
These card styles were outdated:
https://ethereum.org/en/contributing/translation-program/#in-progress so @samajammin I updated them to similar ones like on `/languages` and `/about`.
<!--- This project accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

samajammin commented on src/components/SharedStyledComponents.js:
```diff
@@ -443,3 +445,28 @@ export const dropdownIconContainerVariant = {
+  border-radius: 2px;
+  width: 100%;
+  border: 1px solid ${(props) => props.theme.colors.lightBorder};
+  ${"" /* box-shadow: ${(props) => props.theme.colors.tableBoxShadow}; */}
```
Let's just delete this line given that it's commented out.

samajammin commented on src/components/SharedStyledComponents.js:
```diff
@@ -443,3 +445,28 @@ export const dropdownIconContainerVariant = {
+
+// Common styled item card for languages/about
+
+export const CardItem = styled(GatsbyLink)`
```
I *believe* we can just use `Link` here vs. `GatsbyLink`. In `Link` component we check to see if language path has been explicitly set: https://github.com/ethereum/ethereum-org-website/blob/dev/src/components/Link.js#L131

samajammin commented on src/components/SharedStyledComponents.js:
```diff
@@ -3,6 +3,8 @@ import styled from "styled-components"
 import Card from "./Card"
 import Link from "./Link"
+//Imported for LangItem
+import { Link as GatsbyLink } from "gatsby"
```
I don't think this is needed. See comment below.

samajammin commented on src/components/Roadmap.js:
```diff
@@ -9,6 +9,9 @@ import { FakeLinkExternal, H2, H3 } from "./SharedStyledComponents"
 
 import { translateMessageId } from "../utils/translations"
 
+//Import for Item / Card
```
No need for comment, please remove 😄

samajammin commented on src/components/SharedStyledComponents.js:
```diff
@@ -443,3 +445,28 @@ export const dropdownIconContainerVariant = {
+  text-decoration: none;
+  margin: 1rem 1rem 1rem 0;
+  padding: 1rem;
+  flex: 1 1 200px;
```
I don't think this quite works.

It looks great for the in-progress translation cards:
https://share.getcloudapp.com/yAu6QOkE

But it warps the styles of the language cards (see your preview deploy: https://deploy-preview-2481--ethereumorg.netlify.app/en/languages/): 
https://share.getcloudapp.com/bLug5EBE
Those cards likely need that default `width: 240px;` property. Perhaps you overwrite these base styles in that file (e.g. `LangItem = styled(CardItem)`...

Anuragtech02 replied:
Got it! I'll overwrite it for the language card. Also do let me know if you want to get it changed for the work in progress cards on `/about`.

samajammin reviewed (changes requested):
Thanks for the PR! A couple comments. Please let me know if you have any questions 😄

Anuragtech02 commented:
Thanks @samajammin !, I've made the changes

wackerow commented on src/components/TranslationsInProgress.js:
```diff
@@ -1,33 +1,17 @@
-  text-decoration: none;
-  margin: 1rem 1rem 1rem 0;
-  padding: 1rem;
+const LangItem = styled(CardItem)`
```
This renders with an extra "external" arrow glyph. By switching this to use `CardItem`, it now treats this link differently and adds an arrow glyph. This is already accommodated for in the `FakeExternalLink` which puts the arrow next to the appropriate text.

tl;dr - Add `hideArrow` attribute to L42 to hide extra arrow:
`<LangItem to={url} key={lang.code} hideArrow>`

wackerow commented on src/components/SharedStyledComponents.js:
```diff
@@ -443,3 +443,27 @@ export const dropdownIconContainerVariant = {
+  text-decoration: none;
+  margin: 1rem 1rem 1rem 0;
+  padding: 1rem;
+  flex: 1 1 240px;
```
This causes the bottom row of the Translations page to have ~1-3 cards that get stretched to fill the whole row (depending on screen width).
`flex: 0 1 240px;` should prevent this

wackerow reviewed (changes requested):
@Anuragtech02 Thanks again for the PR =) Spotted two things... one is the redundant external arrow, the other being the bottom row expanding full width (see inline comments). Otherwise should be good to go!

Anuragtech02 commented:
Thanks @wackerow ! I've added the hideArrow attr and set the flex grow to zero. I've also removed a redundant media query which was trying to set card width to 100% (which was already 100%).

samajammin reviewed (changes requested):
Almost there @Anuragtech02!

Looking at your preview deploy (https://deploy-preview-2481--ethereumorg.netlify.app/en/languages/) in mobile:
![Image 2021-03-03 at 2 29 39 PM](https://user-images.githubusercontent.com/8097623/109881303-ebb75d00-7c2c-11eb-8007-588db44412df.jpg)

The cards are no longer full width, like they are in production:
![Image 2021-03-03 at 2 30 15 PM](https://user-images.githubusercontent.com/8097623/109881350-fc67d300-7c2c-11eb-81d7-c265ac5a752f.jpg)

I suspect the media query or the flex grow you deleted is actually needed on the language page items.

Anuragtech02 commented:
> Almost there @Anuragtech02!
> 
> Looking at your preview deploy (https://deploy-preview-2481--ethereumorg.netlify.app/en/languages/) in mobile:
> ![Image 2021-03-03 at 2 29 39 PM](https://user-images.githubusercontent.com/8097623/109881303-ebb75d00-7c2c-11eb-8007-588db44412df.jpg)
> 
> The cards are no longer full width, like they are in production:
> ![Image 2021-03-03 at 2 30 15 PM](https://user-images.githubusercontent.com/8097623/109881350-fc67d300-7c2c-11eb-81d7-c265ac5a752f.jpg)
> 
> I suspect the media query or the flex grow you deleted is actually needed on the language page items.

@samajammin  The media query was not the issue, it was the flex-grow. I added a media query for mobile screens which sets flex grow to 1 which now solves the issue for us.

wackerow reviewed (approved):
👍 @Anuragtech02 Looks good! Added one last little tweak to remove to right margin on mobile screens to even it out. Thanks again!
cc @samajammin

samajammin reviewed (approved):
Thanks for following through on this ❤️

samajammin commented:
@all-contributors please add @Anuragtech02 for code.
